packages/components/src/components/data-list/data-list.tsx

```diff
@@ -0,0 +1,244 @@
+import classNames from 'classnames';
+import React from 'react';
+import { TransitionGroup } from 'react-transition-group';
+import {
+    List as _List,
+    CellMeasurer as _CellMeasurer,
+    CellMeasurerCache,
+    CellMeasurerProps,
+    AutoSizer as _AutoSizer,
+    type AutoSizerProps,
+    ScrollParams,
+    ListProps,
+    ListRowProps,
+} from 'react-virtualized';
+import { isMobile, isDesktop } from '@deriv/shared';
+import DataListCell from './data-list-cell';
+import DataListRow from './data-list-row';
+import ThemedScrollbars from '../themed-scrollbars';
+import { MeasuredCellParent } from 'react-virtualized/dist/es/CellMeasurer';
+
+const List = _List as unknown as React.FC<ListProps>;
+const AutoSizer = _AutoSizer as unknown as React.FC<AutoSizerProps>;
+const CellMeasurer = _CellMeasurer as unknown as React.FC<CellMeasurerProps>;
+type CellType = { Cell: typeof DataListCell };
+export type TRowRenderer = (params: { row: any; is_footer?: boolean; measure?: () => void }) => React.ReactNode;
+type DataListProps = {
+    className?: string;
+    data_source: any;
```

Comment:
@ali-hosseini-deriv could you please declare a type for data_source array? data_source can be different but we have to declare its type to tell TS what properties it must have in order for DataList to work correctly. I might be something like this: TPosition[] | TCryptoTransaction[] | etc 🙂

Reply:
this data source can be anything and we map it using keymapper function that should be passed to the component!
so not sure how we can make a generic type for this.